docs/reference/api.rst

```diff
@@ -241,3 +241,25 @@ Compatibility
 
    to_dict
    from_dict
+
+
+Typing
+======
+
+.. autosummary::
+   :toctree: ../generated
+
+   is_variable
+   is_dataset
+   is_data_array
+   is_dataset_or_array
+   typing.is_scalar
```

Comment:
Not really. I just thought since we have them, we might as well tell the users. To you think they would change meaning at some point without a major change to Variables / Datasets themselves?

What about moving `typing` to the top level? Should it still be 'hidden' in `utils` if we treat these functions as implementation details?